src/Eventso.Subscription.Kafka.DeadLetter.Postgres/PoisonEventStore.cs

```diff
@@ -1,512 +1,281 @@
+        var result = new Dictionary<ConsumingTarget, long>();
+        var reader = await command.ExecuteReaderAsync(token);
+        while (await reader.ReadAsync(token))
+            result.Add(new ConsumingTarget(reader.GetString(0), reader.GetString(1)), reader.GetInt64(2));
+        
+        return result;
     }
 
-    public async Task<long> Count(string topic, CancellationToken cancellationToken)
+    public async Task<long> CountPoisonedEvents(string groupId, string topic, CancellationToken cancellationToken)
     {
-        await using var connection = _connectionFactory.ReadOnly();
+        await using var connection = connectionFactory.ReadOnly();
 
         await using var command = new NpgsqlCommand(
-            "SELECT COUNT(*) FROM eventso_dlq.poison_events WHERE topic = @topic;",
-            connection)
-        {
-            Parameters =
-            {
-                new NpgsqlParameter<string>("topic", topic)
-            }
-        };
+            "SELECT COUNT(*) FROM eventso_dlq.poison_events WHERE topic = @topic AND group_id = @groupId;",
+            connection);
+        command.Parameters.Add(new NpgsqlParameter<string>("topic", topic));
+        command.Parameters.Add(new NpgsqlParameter<string>("groupId", groupId));
 
         await connection.OpenAsync(cancellationToken);
 
         var result = await command.ExecuteScalarAsync(cancellationToken);
         return result != null ? (long)result : 0;
     }
 
-    public async Task<bool> IsStreamStored(string topic, Guid key, CancellationToken token)
+    public async Task<bool> IsKeyPoisoned(string groupId, string topic, byte[] key, CancellationToken token)
     {
-        await using var connection = _connectionFactory.ReadOnly();
+         await using var connection = connectionFactory.ReadOnly();
 
-        await using var command = new NpgsqlCommand(
-            "SELECT TRUE FROM eventso_dlq.poison_events WHERE topic = @topic AND key = @key LIMIT 1;",
-            connection)
-        {
-            Parameters =
-            {
-                new NpgsqlParameter<string>("topic", topic),
-                new NpgsqlParameter<Guid>("key", key)
-            }
-        };
+         await using var command = new NpgsqlCommand(
+             "SELECT TRUE FROM eventso_dlq.poison_events WHERE topic = @topic AND key = @key AND group_id = @groupId LIMIT 1;",
+             connection);
+         command.Parameters.Add(new NpgsqlParameter<string>("topic", topic));
+         command.Parameters.Add(new NpgsqlParameter<ReadOnlyMemory<byte>>("key", key));
+         command.Parameters.Add(new NpgsqlParameter<string>("groupId", groupId));
 
-        await connection.OpenAsync(token);
+         await connection.OpenAsync(token);
 
-        var result = await command.ExecuteScalarAsync(token);
-        return result != null && (bool)result;
+         var result = await command.ExecuteScalarAsync(token);
+         return result != null && (bool)result;
```

Comment:
Should check for DbNull as well

```suggestion
         return result is not (null or DBNull) && (bool)result;
```

another option would be using different query that for the similar query plan will always produce a boolean scalar:
```sql
SELECT EXISTS(
  SELECT 1 FROM eventso_dlq.poison_events
  WHERE topic = @topic AND key = @key AND group_id = @groupId
);
```